test/integration/kind_diagnostics_test.go

```diff
@@ -0,0 +1,69 @@
+//go:build integration_tests
+// +build integration_tests
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	environment "github.com/kong/kubernetes-testing-framework/pkg/environments"
+)
+
+func TestKindDiagnosticDump(t *testing.T) {
+	t.Parallel()
+
+	t.Log("configuring the testing environment")
+	builder := environment.NewBuilder()
+
+	t.Log("building the testing environment and Kubernetes cluster")
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+
+	t.Logf("setting up the environment cleanup for environment %s and cluster %s", env.Name(), env.Cluster().Name())
+	defer func() {
+		t.Logf("cleaning up environment %s and cluster %s", env.Name(), env.Cluster().Name())
+		require.NoError(t, env.Cleanup(ctx))
+	}()
```

Comment:
If I may suggest to move to `t.Cleanup()` instead. Both to stay consistent with other tests and to not fall into the trap of `defer` vs `t.Cleanup` when using `t.Parallel()`

This is not an issue here not, but in case someone adds a subtest in the future it might become a problem.

Ref: https://engineering.mercari.com/en/blog/entry/20220408-how_to_use_t_parallel/

```suggestion
	t.Cleanup(func() {
		t.Logf("cleaning up environment %s and cluster %s", env.Name(), env.Cluster().Name())
		require.NoError(t, env.Cleanup(ctx))
	})
```

Reply:
Fully agree, thanks, done!